cli/command/container/run.go

```diff
@@ -68,7 +68,7 @@ func NewRunCommand(dockerCli command.Cli) *cobra.Command {
 }
 
 func runRun(dockerCli command.Cli, flags *pflag.FlagSet, ropts *runOptions, copts *containerOptions) error {
-	proxyConfig := dockerCli.ConfigFile().ParseProxyConfig(dockerCli.Client().DaemonHost(), copts.env.GetAll())
+	proxyConfig := dockerCli.ConfigFile().ParseProxyConfig(dockerCli.Client().DaemonHost(), opts.ConvertKVStringsToMapWithNil(copts.env.GetAll()))
```

Comment:
https://github.com/docker/cli/pull/1573 was merged; can you rebase, and update `runCreate()` as well? https://github.com/docker/cli/blob/91bc4ddde205fca60c9932322b34c4808fd0733c/cli/command/container/create.go#L64-L66